stdlib/stdlib.mli

```diff
@@ -188,20 +188,25 @@ val max : 'a -> 'a -> 'a
     The result is unspecified if one of the arguments contains
     the float value [nan]. *)
 
-external ( == ) : 'a -> 'a -> bool = "%eq"
-(** [e1 == e2] tests for physical equality of [e1] and [e2].
+external phys_equal : 'a -> 'a -> bool = "%eq"
+(** [phys_equal e1 e2] tests for physical equality of [e1] and [e2].
    On mutable types such as references, arrays, byte sequences, records with
    mutable fields and objects with mutable instance variables,
-   [e1 == e2] is true if and only if physical modification of [e1]
+   [phys_equal e1 e2] is true if and only if physical modification of [e1]
    also affects [e2].
-   On non-mutable types, the behavior of [( == )] is
+   On non-mutable types, the behavior of [phys_equal] is
    implementation-dependent; however, it is guaranteed that
-   [e1 == e2] implies [compare e1 e2 = 0].
+   [phys_equal e1 e2] implies [compare e1 e2 = 0].
+   @since 5.4
+*)
+
+external ( == ) : 'a -> 'a -> bool = "%eq"
+(** Operator alias to {!Stdlib.phys_equal}.
```

Comment:
How about adding the `@deprecated` odoc tag to `( == )` and `( != )`: https://ocaml.github.io/odoc/odoc/odoc_for_authors.html#signature-tags

This would not break anyone's build since it is documentation only, however it would let people know to use `phys_equal` instead.

Reply:
Although this is a good idea, no deprecation has been discussed yet.

I propose to add this in a follow-up PR. I'm afraid trying to push too much things get things stuck again

Reply:
I thought the deprecation plan was just discussed above, @kit-ty-kate 's comment. Note that I am not suggesting to add the `@@deprecated` attribute which might break people's builds. I am suggesting only adding a `@deprecated ...` odoc tag which would inform users to prefer `phys_equal`. While they look syntactically similar, they are in totally different contexts. The latter is inside a comment so it has no impact on anyone's build.

Reply:
Having a documentation note as suggested by @yawaramin about planned deprecation seems like a good temporary solution to me (in addition to what @kit-ty-kate was saying).